rewrite-java/src/main/java/org/openrewrite/java/search/SemanticallyEqual.java

```diff
@@ -871,7 +871,7 @@ public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, J j)
                     return method;
                 }
 
-                boolean static_ = method.getMethodType().hasFlags(Flag.Static);
+                boolean static_ = method.getMethodType() != null && method.getMethodType().hasFlags(Flag.Static);
                 J.MethodInvocation compareTo = (J.MethodInvocation) j;
                 if (!method.getSimpleName().equals(compareTo.getSimpleName()) ||
                     !TypeUtils.isOfType(method.getMethodType(), compareTo.getMethodType()) ||
```

Comment:
There's another instance of `compareTo.getMethodType().hasFlags(Flag.Static)` just below this line; would that not be similarly affected?

Reply:
I have been wondering about that one myself too. I'm still fighting with the gradle build (company machine behind proxy), so I cannot easily test it. I can use the patched version at home to see if there is more to fix, but that might take a day or two.